x-pack/legacy/plugins/maps/server/maps_telemetry/maps_telemetry.js

```diff
@@ -88,24 +99,23 @@ export function buildMapsTelemetry(savedObjects, settings) {
   };
 }
 
-async function getSavedObjects(savedObjectsClient) {
-  const gisMapsSavedObject = await savedObjectsClient.find({
-    type: MAP_SAVED_OBJECT_TYPE
-  });
-  return _.get(gisMapsSavedObject, 'saved_objects');
+async function getMapSavedObjects(savedObjectsClient) {
+  const mapsSavedObjects = await savedObjectsClient.find({ type: MAP_SAVED_OBJECT_TYPE });
+  return _.get(mapsSavedObjects, 'saved_objects', []);
+}
+
+async function getIndexPatternSavedObjects(savedObjectsClient) {
+  const indexPatternSavedObjects = await savedObjectsClient.find({ type: 'index-pattern' });
```

Comment:
You need to set `per_page` and page through results. Otherwise, you are only going to gets stats on first 20 index-patterns. https://github.com/elastic/kibana/blob/master/src/core/server/saved_objects/service/lib/repository.ts#L417

The same is true when fetching map saved objects.

Reply:
Aha, didn't know that, thanks!

Let's address in separate PR https://github.com/elastic/kibana/issues/51345